src/hatti/views/dataview.cljs

```diff
@@ -88,6 +88,39 @@
                              {:selected view})
       (put! shared/event-chan {:re-render view}))))
 
+(defn activate-settings-view! [view settings-section]
+  (let [view (keyword view)
+        settings-section (keyword settings-section)
+        views (-> @shared/app-state :views :all)
+        settings-views (-> @shared/app-state :views :settings :all)]
+    (when (contains? (set settings-views) settings-section)
+      (merge-into-app-state! shared/app-state
+                             [:views]
+                             {:selected view})
+      (merge-into-app-state! shared/app-state
+                             [:views :settings]
+                             {:active-tab settings-section})
+      (put! shared/event-chan {:re-render view}))))
+
+(defn activate-integrated-apps-view! [view settings-section app-type]
+  (let [view (keyword view)
+        settings-section (keyword settings-section)
+        settings-views (-> @shared/app-state :views :settings :all)]
+    (when (contains? (set settings-views) settings-section)
+      (transact!
+       shared/app-state
+       (fn [app-state]
+         (-> app-state
+             (assoc-in [:views] {:selected view})
```

Comment:
I think we should avoid overwriting `:views` as we may be removing keys nested under `:views` that are in used  elsewhere. e.g. `:all` https://github.com/onaio/hatti/blob/cd78137b5bff78cf8c6250089a97b030568df280/src/hatti/shared.cljs#L40

I think we should leave it as is but change the first `assoc-in` to `(assoc-in [:views :selected] view)` to avoid it from overwriting `:views`